cpp/oneapi/dal/backend/primitives/ndarray.hpp

```diff
@@ -539,14 +712,26 @@ inline sycl::event fill(sycl::queue& q,
 }
 #endif
 
+/// Multidimensional array
```

Comment:
How does it differ from sycl's own class?

Reply:
I do not think sycl has own multidimensional data representations.

Reply:
But it does have a 1D `marray`. Lots of the usage for views is as 1D containers, so I was curious about their usage over sycl's.

Reply:
I think besides of multidimensionality, our data structures are interoperable in the sense one can easily view the same data block as dal::table, dal::array, dal::backend::primitives::ndview, etc.

Also here we implement convenience functions like slice, transpose, to_host, etc.

With sycl own structures the amount of code inside oneDAL algorithmic kernels would be much bigger. Anyway we would need to wrap some common operations into functions. This class groups those common operations on the data together.

Reply:
Would be helpful to add a hint to prefer these custom classes instead of sycl's.